File: 프로그래머스/lv2/42587. 프린터/프린터.py
def solution(priorities, location):
    answer = 0
    
    data = []
    for idx, priority in enumerate(priorities):
      data.append(( priority , idx ))

    
    while data:
      temp = data.pop(0)
      
      flag = False
      for i in data: # 뺄수 없는 경우
        if temp[0] < i[0]:
          data.append(temp)
          flag = True
          break;

      #뺄수 없는 경우
      if (flag):
        continue
      else:
        answer += 1
        if temp[1] == location:
          break;
      
    return answer

File: src/programmers/implement/프린터.java
package programmers.implement;

import java.util.LinkedList;
import java.util.Queue;

public class 프린터 {

    static class Item {
        int value;
        int index;

        public Item(int value, int index) {
            this.value = value;
            this.index = index;
        }
    }

    public static int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Item> q = new LinkedList<>();

        for(int i=0; i< priorities.length; i++) {
            q.add(new Item(priorities[i], i));
        }

        while(!q.isEmpty()) {
            Item item = q.poll();
            boolean flag = false;

            for(Item i : q) {
                if(item.value < i.value) {
                    flag = true;
                    break;
                }
            }

            if(flag) {
                q.add(item);
            } else { //우선순위가 제일 큼?
                answer++;
                if(item.index == location) {
                    return answer;
                }
            }
        }
        return 0;
    }
}
